packages/preservation_spine/tests/test_seal_restore.py

```
@@ -0,0 +1,1250 @@
+        if path == target and not substituted:
+            target.rename(preserved)
+            try:
+                os.link(outside, target)
+            except OSError as exc:
+                pytest.skip(f"hardlink substitution unavailable: {exc}")
+            substituted = True
+        return original_hash(path)
+
+    monkeypatch.setattr(seal_module, "_hash_regular_file", substitute_before_hash)
+
+    with pytest.raises(CapsuleVerificationError, match="hardlink|changed"):
```

Ruff (0.16.1)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)